modules/exploits/linux/http/eyesofnetwork_autodiscovery_rce.rb

```diff
@@ -71,29 +100,149 @@ def server_addr
 
   def check
     vprint_status("Running check")
-    res = send_request_cgi 'uri' => normalize_uri(target_uri.path, '/eonapi/getApiKey')
+    res = send_request_cgi 'uri' => normalize_uri(target_uri.path,'/login.php')
 
     unless res
       return CheckCode::Unknown('Connection failed')
     end
 
-    unless res.code == 401 && res.body.include?('api_version')
+    unless res.code == 200 && res.body.include?('EyesOfNetwork')
       return CheckCode::Safe('Target is not an EyesOfNetwork application.')
     end
 
-    version = res.get_json_document()['api_version'] rescue ''
+    res_api = send_request_cgi 'uri' => normalize_uri(target_uri.path, '/eonapi/getApiKey')
+
+    unless res_api
+      return CheckCode::Unknown('Connection failed')
+    end
+
+    if res_api.code == 404 #The EON api was only introduced in EON 5.2
+      @version = '5.1'
+      return CheckCode::Appears("Target is EyesOfNetwork version 5.1 or older.")
+    end
 
-    if version.to_s.eql? ''
+    unless res_api.code == 401 && res_api.body.include?('api_version')
+      return CheckCode::Safe('Target is not an EyesOfNetwork application.')
+    end
+
+    api_version = res_api.get_json_document()['api_version'] rescue ''
+
+    if api_version.to_s.eql? ''
```

Comment:
It's not really necessary to use the `.eql?` string method, you can simply do `api_version.to_s == ''` instead and it's a bit more clear.